### PR TITLE
Updated LICENSE and AUTHOR info regarding xdrfile (closes #679)

### LINKED ISSUE #679: New xdrlib information in AUTHORS and LICENSE is missing/outdated
I'm opening this as a reminder so as not to let 0.14.0 ship without these fixes.

The new license aspects of xdrlib are outdated. The Gromacs base is now BSD.
We also haven't decided if the name libxdrfile2 stays, or if we call our modifications something else: with @kain88-de's cython approach (#441) and splitting of `xtc_seek.c`/`trr_seek.c` from `xdrfile.c` (#661) there is less of a clear new library to speak of.
@orbeckst has advocated keeping the name. I agree with that.
Right now we don't provide an actual `libxdrfile2.so` or `libxdrfile2.a`, but rather a mix of that with cython bindings (called `xdrlib.so`). This gets confusing but `xdrlib.so` could be renamed `libxdrfile2` to keep the external appearance and conform to the name we were using before.

### PR DESCRIPTION
**Pull request by mnmelo**
Rebranded our xdrlib with cython bindings as libmdaxdr.

Further standardized XDR error message reporting and exceptions
(now mostly IOError).

Removed reference to the old SWIG numpy.i in the credits.

Made logo LICENSE info consistent

### CODE REVIEW

File: package/CHANGELOG [hunk 1]
```diff
@@ -34,6 +34,7 @@ Enhancement
 
 Changes
 
+  * xdrlib was rebranded limdaxdr. (Issue #679)
```
Comment (kain88-de) [inline package/CHANGELOG]: Libmdaxdr

**Comment by kain88-de** [comment]
@mnmelo you forgot to add the renamed c-file. This is the reason the minimal build fails.  The full build fails because you didn't change the offsets detection code in  `XDR.py`

**Comment by mnmelo** [comment]
Ouch... That was a sloppy rebasing I made. And dumb to have skipped libmdaxdr.c (had just finished working on skipping the cythonized files).

**Comment by orbeckst** [comment]
Overall looks good to me, thanks for doing the legalese.
I am assigning the merge to @kain88-de as he's patrolling here anyway.

**Pull request merged**